File: results/part4/Calculator/android/app/src/main/java/com/calculator/CalculatorModule.kt
package com.calculator

import android.util.Log
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class CalculatorModule(reactContext:ReactApplicationContext):ReactContextBaseJavaModule(reactContext) {
    override fun getName() = "CalculatorModule"

    @ReactMethod
    fun executeCalc(action:String, numberA:Double, numberB:Double, promise: Promise){
        Log.v("CalculatorTag", "give action is "+ action)
        if(action == "plus"){
            promise.resolve(numberA + numberB);
            return;
        }

        if(action == "minus"){
            promise.resolve(numberA - numberB);
            return;
        }

        if(action == "multiply"){
            promise.resolve(numberA * numberB);
            return;
        }

        if(action == "divide"){
            promise.resolve(numberA/numberB);
            return;
        }


        promise.reject("Unexpected action type" + action)
    }
}

File: results/part4/Calculator/ios/CalculatorModule.swift
//
//  CalculatorModule.swift
//  Calculator
//
//  Created by Pratt Yeon on 2022/12/10.
//

import Foundation

@objc(CalculatorModule)
class CalculatorModule:NSObject {
  
  @objc(executeCalc:numberA:numberB:resolver:rejector:)
  public func executeCalc(_ action:String, numberA:Int, numberB:Int, resolver:RCTPromiseResolveBlock, rejector:RCTPromiseRejectBlock) -> Void{
    
    if(action == "plus"){
      resolver(numberA + numberB);
      return;
    }
    
    if(action == "minus"){
      resolver(numberA-numberB);
      return;
    }
    
    if(action == "multiply"){
      resolver(numberA * numberB);
      return;
    }
    
    if(action == "divide"){
      resolver(numberA / numberB);
      return;
    }
    
    rejector("Unexpected action type", action, nil);
    
  }
  
}
